src/NonceManager.sol

```diff
@@ -2,32 +2,33 @@
 pragma solidity ^0.8.23;
 
 import {INonceManager} from "./interfaces/INonceManager.sol";
-import {MinimalDelegationStorage, MinimalDelegationStorageLib} from "./libraries/MinimalDelegationStorage.sol";
 
 /// @title NonceManager
 /// @notice A contract that manages nonces to prevent replay attacks
 abstract contract NonceManager is INonceManager {
+    mapping(uint256 key => uint256 seq) nonceSequenceNumber;
+
     /// @dev Must be overridden by the implementation
     function _onlyThis() internal view virtual {}
 
     /// @inheritdoc INonceManager
     function getNonce(uint256 key) external view override returns (uint256 nonce) {
-        return MinimalDelegationStorageLib.get().nonceSequenceNumber[uint192(key)] | (key << 64);
+        return nonceSequenceNumber[key] | (key << 64);
```

Comment:
hm this is not super easy to understand and I'm realizing now that the documentation is incorrect here.

the `key` parameter is a uint256 but only the lower 192 bits are used since it is just a key, so we can directly use the uint256 value, but I think we should mask the upper 64 bits before keying the mapping.

then we have to shift the key over 64 to return a nonce in the format of [key][nonceNumber] packed.

Maybe we can use some consistency in naming in the nonce manager to understand what values we are working with

key - 192 bits
packedNonce - 256 bits consists of key and nonceNumber
nonceNumber - 64 bits

Reply:
ah yeah the logic must have gotten mixed up in handling merge conflicts. I can update the naming

Reply:
wait i thought only the upper 192 bits are used for the key and the rest is the seq?

Reply:
the upper 192 bits are used for the key when it is passed in as a packed nonce [key]|[seq]

here it is just being passed as a key, so its a uint256 but only the lower 192 bits are used, so we need to mask the upper bits

Reply:
also i think we should change this to just return the sequence